test/hpke_test.c

```diff
@@ -0,0 +1,1432 @@
+        return 0;
+    /* suggested suite with P-521, just to be sure we hit long values */
+    enclen = 200; /* reset, 'cause get_grease() will have set for suite2  */
+    suite.kem_id = OSSL_HPKE_KEM_ID_P521;
+    if (!TEST_true(OSSL_HPKE_get_grease_value(NULL, NULL, &suite, &suite2,
+                                              enc, &enclen, ct, ctlen)))
+        return 0;
+    enclen = 200;
+    ctlen = 2; /* too-short cttext (can't fit an aead tag) */
+    if (!TEST_false(OSSL_HPKE_get_grease_value(NULL, NULL, NULL, &suite2,
+                                               enc, &enclen, ct, ctlen)))
+        return 0;
+    ctlen = 500;
+    enclen = 200;
+    suite.kem_id = OSSL_HPKE_KEM_ID_X25519; /* back to default */
+    suite.aead_id = 0x1234; /* bad aead */
+    if (!TEST_false(OSSL_HPKE_get_grease_value(NULL, NULL, &suite, &suite2,
+                                               enc, &enclen, ct, ctlen)))
+        return 0;
+    enclen = 200;
+    suite.aead_id = def_suite.aead_id; /* good aead */
+    suite.kdf_id = 0x3451; /* bad kdf */
+    if (!TEST_false(OSSL_HPKE_get_grease_value(NULL, NULL, &suite, &suite2,
+                                               enc, &enclen, ct, ctlen)))
+        return 0;
+    enclen = 200;
+    suite.kdf_id = def_suite.kdf_id; /* good kdf */
+    suite.kem_id = 0x4517; /* bad kem */
+    if (!TEST_false(OSSL_HPKE_get_grease_value(NULL, NULL, &suite, &suite2,
+                                               enc, &enclen, ct, ctlen)))
+        return 0;
+    return 1;
+}
+
+static int test_hpke_ikms(void)
+{
+    int res = 1;
+
+    res = test_hpke_one_ikm_gen(0x20,
+                                ikm25519, sizeof(ikm25519),
+                                pub25519, sizeof(pub25519));
+    if (res != 1)
+        return res;
+
+    res = test_hpke_one_ikm_gen(0x12,
+                                ikmp521, sizeof(ikmp521),
+                                pubp521, sizeof(pubp521));
+    if (res != 1)
+        return res;
+
+    res = test_hpke_one_ikm_gen(0x10,
+                                ikmp256, sizeof(ikmp256),
+                                pubp256, sizeof(pubp256));
+    if (res != 1)
+        return res;
+
+    res = test_hpke_one_ikm_gen(0x10,
+                                ikmiter, sizeof(ikmiter),
+                                pubiter, sizeof(pubiter));
+    if (res != 1)
+        return res;
+
+    return res;
+}
+int setup_tests(void)
+{
+    ADD_TEST(x25519kdfsha256_hkdfsha256_aes128gcm_base_test);
+    ADD_TEST(x25519kdfsha256_hkdfsha256_aes128gcm_psk_test);
+    ADD_TEST(P256kdfsha256_hkdfsha256_aes128gcm_base_test);
+    ADD_TEST(export_only_test);
+    ADD_TEST(test_hpke_export);
+    ADD_TEST(test_hpke_modes_suites);
+    ADD_TEST(test_hpke_suite_strs);
+    ADD_TEST(test_hpke_grease);
+    ADD_TEST(test_hpke_ikms);
+    ADD_TEST(test_hpke_random_suites);
+    return 1;
+}
+void cleanup_tests(void)
+{
+}
```

Comment:
This can be deleted. It doesn't need to be present if there is no cleanup to be done.

Reply:
done